tennis-scheduler-web-app/src/components/TennisCourtAddChange.jsx

```diff
@@ -0,0 +1,165 @@
+        const { name, value } = e.target;
+        setAddress((address) => ({ ...address, [name]: value }));
+    };
+
+    const onImageChange = e => {
+        const tennisCourtImage = e.target.value.split("fakepath")[1].substring(1);
+        setImage(tennisCourtImage)
+        setTennisCourt(tennisCourt => ({
+            ...tennisCourt,
+            image: tennisCourtImage
+        }))
+        setCheck(true)
+    }
+
+    const onSubmit = e => {
+        e.preventDefault()
+        setTennisCourt(tennisCourt => ({
+            ...tennisCourt,
+            address: address
+        }))
+        setTennisCourt(tennisCourt => ({
+            ...tennisCourt,
+            surfaceType: surface
+        }))
+        setFormErrors(validation(tennisCourt))
+        if (Object.keys(formErrors).length === 0 && !emptyForm) {
+            if (id)
+                tennisCourtChange();
+            else
+                tennisCourtAdd();
+        }
+    }
+
+    const tennisCourtChange = () => {
+        changeTennisCourt(tennisCourt).then(() => {
+            navigate('/');
+            window.location.reload()
+        }).catch(() =>
+            toast.error("Something went wrong, try again!", { position: toast.POSITION.BOTTOM_CENTER }))
+    }
+
+    const tennisCourtAdd = () => {
+        addTennisCourt(tennisCourt).then(() => {
+            navigate('/');
+            window.location.reload()
+        }).catch(() => toast.error("Something went wrong, try again!", { position: toast.POSITION.BOTTOM_CENTER }))
+    }
+
+    const validation = (court) => {
+        const errors = {};
+        const lettersRegex = new RegExp(/^$|^[A-Za-z ]+$/);
+        const streetNumberRegex = new RegExp("^\\d+$");
+        if (!court.name) errors.name = "Name is required!";
+        if (!court.description) errors.description = "Description is required";
+        if (!court.surfaceType) errors.surfaceType = "You have to select surface type!";
+        if (!lettersRegex.test(court.address.country)) errors.country = "Only letters are allowed!"
+        if (!lettersRegex.test(court.address.city)) errors.city = "Only letters are allowed!"
+        if (!lettersRegex.test(court.address.street)) errors.street = "Only letters are allowed!"
+        setEmptyForm(false);
+        return errors;
+    };
+
+    const handleChangeSurfaceType = e => {
+        setSurface(e.target.value)
+        setChangedType(e.target.value)
+    }
+
+    const onCancel = e => navigate('/')
+
+    useEffect(() => {
+        if (id)
+            getTennisCourt(id).then(
+                response => {
+                    setTennisCourt(response.data)
+                    setAddress(response.data.address)
+                    setCheck(true)
+                    setImage(response.data.image)
+                    setChangedType(response.data.surfaceType)
+                    setSurface(response.data.surfaceType)
+                }
+            ).catch(
+                () => toast
```

Comment:
Why is it empty?

Reply:
Because I don't do anything with response, it shows this generic response that something went wrong.